### PR TITLE
Backend FastApi

### PR DESCRIPTION
**Pull request by kumarpanzade**

### CODE REVIEW

File: service/audio_service.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+# Import configurations and functions from modules
+from config import openai_api_key, model_id, model_path
+from load_model import load_model
+
+# Load environment variables
+load_dotenv()
+
+openai.api_key = openai_api_key
+#Load whisher model
+print("model path -> ",model_path)
```
Comment (sourabh-josh) [inline service/audio_service.py:18]: if we have set up on logger then why not use it where as well ?

File: service/download_model.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import sys
+
+# # Check if two command-line arguments are provided
+if len(sys.argv) !=3:
+    print("Usage: python download_model.py <whisper_model_id> <whisper_model_output_path>")
```
Comment (sourabh-josh) [inline service/download_model.py:5]: here as well think if we could use logger
also the print messages doesn't actually informative

File: service/extract_entities.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            model="gpt-4o",
+            messages=[
+                {"role": "system", "content": "You are a helpful assistant that extracts information from Indian multilingual text."},
+                {"role": "user", "content": prompt}
+            ],
+            max_tokens=500
+        )
+        response_text = response.choices[0].message.content
+    except Exception as e:
+        return f"Error during OpenAI API call: {e}", "Detailed view not available."
```
Comment (sourabh-josh) [inline service/extract_entities.py:45]: add an error log point here

File: service/main.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+            raise HTTPException(status_code=400, detail="Invalid file type")
+
+        # Create a temporary file to save the uploaded file
+        with tempfile.NamedTemporaryFile(delete=False, suffix=os.path.splitext(audioFile.filename)[1]) as temp_file:
+            temp_file_path = temp_file.name
+            # Save the file
+            with open(temp_file_path, "wb") as buffer:
+                buffer.write(await audioFile.read())
+
+        #translation = translate_with_ollama(transcription)
```
Comment (sourabh-josh) [inline service/main.py:41]: what model are we using check
it says ollama in comments and openai in function call

File: service/main.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        translation = translate_with_whisper(temp_file_path)
+
+        logger.info("translation done")
+
+        # Clean up the temporary file
+        os.remove(temp_file_path)
+        
+        #summary = summarize_using_llama(translation)
+        summary = summarize_using_openai(translation)
```
Comment (sourabh-josh) [inline service/main.py:50]: same here

File: service/main.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+        # Check file type
+        if not audioFile.filename.endswith(('m4a', 'mp4','mp3','webm','mpga','wav','mpeg')):
+            logger.error("invalid file type")
+            raise HTTPException(status_code=400, detail="Invalid file type")
+
+        # Create a temporary file to save the uploaded file
+        with tempfile.NamedTemporaryFile(delete=False, suffix=os.path.splitext(audioFile.filename)[1]) as temp_file:
+            temp_file_path = temp_file.name
+            # Save the file
```
Comment (sourabh-josh) [inline service/main.py:37]: do we need to save this file locally 
please check if we do it using io lib
```

        file_content = await audioFile.read()
        file_like_object = io.BytesIO(file_content)

        translation = translate_with_whisper(file_like_object)
```

**Review by sourabh-josh — COMMENTED** [review]
added comments please check

File: service/main.py (hunk 6 above)
Comment (SharyuMarwadi) [reply service/main.py:37]: The model expect file path input not file object. Hence, this does not work

Comment (SharyuMarwadi) [reply service/main.py]: The api request body is now changed to uploaded audio file link

File: service/main.py (hunk 5 above)
Comment (SharyuMarwadi) [reply service/main.py:42]: Updated

File: service/main.py (hunk 4 above)
Comment (SharyuMarwadi) [reply service/main.py]: updated

**Review by sourabh-josh — APPROVED** [review]

**Pull request merged**